src/payment/strategies/googlepay/googlepay.ts

```diff
@@ -0,0 +1,201 @@
+        totalPrice: string;
+        totalPriceStatus: string;
+    };
+}
+
+export interface GooglePayIsReadyToPayResponse {
+    result: boolean;
+    paymentMethodPresend?: boolean;
+}
+
+export interface GooglePaySDK {
+    payments: {
+        api: {
+            PaymentsClient: {
+                new(options: GooglePayPaymentOptions): GooglePayClient;
+            },
+        },
+    };
+}
+
+export enum ButtonType {
+    long = 'long',
+    short = 'short',
+}
+export enum ButtonColor {
+    default = 'default',
+    black = 'black',
+    white = 'white',
+}
+
+export interface GooglePayClient {
+    isReadyToPay(options: object): Promise<GooglePayIsReadyToPayResponse>;
+    loadPaymentData(paymentDataRequest: any): Promise<GooglePaymentData>;
+    createButton(options: { [key: string]: string | object }): HTMLElement;
+}
+
+export interface GooglePayHostWindow extends Window {
+    google?: GooglePaySDK;
+}
+
+export interface TokenizePayload {
+    nonce: string;
+    details: {
+        cardType: string;
+        lastFour: string;
+        lastTwo: string;
+    };
+    description: string;
+    type: TokenizeType;
+    binData: {
+        commercial: string;
+        countryOfIssuance: string;
+        debit: string;
+        durbinRegulated: string;
+        healthcare: string;
+        issuingBank: string;
+        payroll: string;
+        prepaid: string;
+        productId: string;
+    };
+}
+
+export interface GooglePaymentData {
+    cardInfo: {
+        cardClass: string;
+        cardDescription: string;
+        cardDetails: string;
+        cardImageUri: string;
+        cardNetwork: string;
+        billingAddress: GooglePayAddress;
+    };
+    paymentMethodToken: {
+        token: string;
+        tokenizationType: string;
+    };
+    shippingAddress: GooglePayAddress;
+    email: string;
+}
+
+export interface GooglePayAddress {
+    address1: string;
+    address2: string;
+    address3: string;
+    address4: string;
+    address5: string;
+    administrativeArea: string;
+    companyName: string;
+    countryCode: string;
+    locality: string;
+    name: string;
+    postalCode: string;
+    sortingCode: string;
+    phoneNumber: string;
+}
+
+export interface PaymentSuccessPayload {
+    email: string;
+    tokenizePayload: TokenizePayload;
+    billingAddress: GooglePayAddress;
+    shippingAddress: GooglePayAddress;
+}
+
+export interface GooglePaymentsError {
+    statusCode: string;
+    statusMessage?: string;
+}
+
+export default function mapGooglePayAddressToRequestAddress(address: GooglePayAddress, id?: string): AddressRequestBody | BillingAddressUpdateRequestBody {
+    return {
+        id,
+        firstName: address.name.split(' ').slice(0, -1).join(' '),
+        lastName: address.name.split(' ').slice(-1).join(' '),
+        company: address.companyName,
+        address1: address.address1,
+        address2: address.address2 + address.address3 + address.address4 + address.address5,
```

Comment:
Is it okay to just concatenate the addresses like this? Do you need a separator?

Reply:
I concatenate because google pay returns five lines of address (unless I didn't added), and to map with BC Address, I pass address line 1 to address1, and the others address lines as address 2